src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -270,42 +270,31 @@ trait SubmissionComponent {
             // attach the input resolutions to the workflow object
             val workflowResolutions: Seq[SubmissionValidationValue] = groupedResolutions.get(wr.id).map { seq =>
 
-              //collect up the workflow resolution results by input
-              val resolutionsByInput = seq.collect {
+              //collect up the workflow resolution results
+              val resolutions = seq.collect {
                 case WorkflowAndInputResolutionRawSqlQuery.WorkflowInputResolutionListResult(workflow, Some(resolution), attribute) =>
-                  (resolution, attribute)
-              }.groupBy { case (resolution, attribute) => resolution.inputName }
-
-              //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
-              //To get around this, we split by input, so that each input is successful (or not) individually.
-              val submissionValues = resolutionsByInput map { case (inputName, recTuples: Vector[(SubmissionValidationRecord, Option[SubmissionAttributeRecord])]) =>
-                val attr = if( recTuples.forall { case (submissionRec, attrRecOpt) => attrRecOpt.isDefined } ) {
-                  //all attributes are real
-                  Some(
-                    submissionAttributeQuery.unmarshalAttributes( recTuples map { case (rec, attrOpt) => ((wr.id, attrOpt.get), None) } ).get(wr.id).get(inputName)
-                  )
-                } else {
-                  None
-                }
-                //assuming that the first elem has the error here
-                SubmissionValidationValue(attr, recTuples.head._1.errorText, inputName)
-              }
-              submissionValues.toSeq
+                  (resolution, attribute) }
+              workflowQuery.unmarshalOneWorkflowInputs(resolutions, workflowId)
```

Comment:
the bones of this function is what (correctly) load workflows from their list of input resolutions, so i pulled it out into `workflowQuery.unmarshalOneWorkflowInputs`